test/test_integration.py

```diff
@@ -528,3 +528,48 @@ def update_graph(n_clicks):
         button.click()
         time.sleep(2)
         self.snapshot('candlestick - 2 click')
+
+    def test_confirm(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            html.Button(id='button', children='Send confirm', n_clicks=0),
+            dcc.ConfirmDialog(id='confirm', message='Please confirm.'),
+            html.Div(id='confirmed')
+        ])
+
+        @app.callback(Output('confirm', 'send_confirm'), [Input('button', 'n_clicks')])
+        def on_click_confirm(n_clicks):
+            if n_clicks:
+                return True
+
+        @app.callback(Output('confirmed', 'children'),
+                      [Input('confirm', 'n_clicks'), Input('confirm', 'submit_n_clicks'), Input('confirm', 'cancel_n_clicks')],)
+        def on_confirmed(n_clicks, submit_n_clicks, cancel_n_clicks):
+            if not n_clicks:
+                return
+            if n_clicks == 1:
+                return 'confirmed'
+            elif n_clicks == 2:
+                return 'canceled'
+
+        self.startServer(app)
+        button = self.wait_for_element_by_css_selector('#button')
+        self.snapshot('confirmation -> initial')
+        time.sleep(1)
+        button.click()
+        time.sleep(1)
+
+        self.driver.switch_to.alert.accept()
+        self.wait_for_text_to_equal('#confirmed', 'confirmed')
+
+        self.snapshot('confirmation -> confirmed')
+
+        time.sleep(0.2)
+        button.click()
+        time.sleep(1)
+        self.driver.switch_to.alert.dismiss()
+        time.sleep(0.5)
+        self.wait_for_text_to_equal('#confirmed', 'canceled')
+
+        self.snapshot('confirmation -> canceled')
```

Comment:
Very nice tests! I'd like to see a couple of other things:
- A similar test for the `dcc.ConfirmDialogProvider`
- Let's test that the callback is called an appropriate number of times. In other tests, I use a thread/multiprocess-safe variables to do that, see e.g. the "`call_count`" variables here: https://github.com/plotly/dash-renderer/blob/master/tests/test_render.py#L478-L484

Reply:
I added a test for the provider and put a call count variable. The call_count get increased normally in my local tests but in circleci it get increased by 6 instead of two, I put the value to be the n_clicks instead for the tests to work on circleci but I think there's an error.